core/arch/arm/plat-k3/drivers/ti_sci.c

```diff
@@ -136,22 +138,26 @@ static inline int ti_sci_do_xfer(struct ti_sci_xfer *xfer)
 	struct k3_sec_proxy_msg *msg = &xfer->tx_message;
 	int ret = 0;
 
+	mutex_lock(&ti_sci_mutex_lock);
+
 	/* Send the message */
 	ret = k3_sec_proxy_send(msg);
 	if (ret) {
 		EMSG("Message sending failed (%d)", ret);
-		return ret;
+		goto unlock;
 	}
 
 	/* Get the response */
 	ret = ti_sci_get_response(xfer);
 	if (ret) {
 		if ((TEE_Result)ret != TEE_ERROR_ACCESS_DENIED)
 			EMSG("Failed to get response (%d)", ret);
-		return ret;
+		goto unlock;
 	}
 
-	return 0;
+unlock:
+	mutex_unlock(&ti_sci_mutex_lock);
+	return ret;
```

Comment:
By the way and out of the scope of this change: I see `k3_sec_proxy_xxx()` return `TEE_Result` values while caller use `int`. Likely something to be fixed at some point of time.